From 6f486513f05a0efb097c510e649cba1322dd2095 Mon Sep 17 00:00:00 2001
From: Fedor Isakov <fedor.isakov@jetbrains.com>
Date: Thu, 25 Jun 2026 09:54:16 +0200
Subject: [PATCH 02/14] MRI-4597 community: reduce default heap size for JVM
 compilation to 10 GB

The previously used unrealistic 20 GB value was causing MPS platform builds to fail.
---
 community/build/jvm-rules/jvm-args.bzl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/community/build/jvm-rules/jvm-args.bzl b/community/build/jvm-rules/jvm-args.bzl
index 4bb4269ba6311..7eb822b03f35e 100644
--- a/community/build/jvm-rules/jvm-args.bzl
+++ b/community/build/jvm-rules/jvm-args.bzl
@@ -12,7 +12,7 @@ def get_jvm_flags(flags):
         #"-XX:+ZGenerational",
         #"-XX:ZUncommitDelay=10",   # reclaim unused memory in 10 secs
         #"-XX:SoftMaxHeapSize=2g",  # try to keep heap compact under 2 Gbytes
-        "-Xmx20g",
+        "-Xmx10g",
         "-Xms256m",
 
         # IJ PSI cache
-- 
2.55.0

